nbclassic/shim.py

```diff
@@ -49,7 +60,49 @@ def _shim_notebookapp_config(
             raise TraitError("Trait not found.")
 
     def shim_config_from_notebook_to_jupyter_server(self, config):
-        """Reorganize a config object that might have outdated traits.
+        """Reorganizes a config object to reroute traits to their expected destinations
+        after the transition from NotebookApp to ServerApp.
+
+        ---- TL;DR ---------------
+
+        A detailed explanation of how traits are handled:
+
+        1. If the argument is prefixed with `ServerApp`,
+            pass this trait to `ServerApp`.
+        2. If the argument is prefixed with `NotebookApp`,
+            - If the argument is a trait of `NotebookApp` *and* `ServerApp`:
+                1. Raise a warning—**for the extension developers**—that
+                    there's redundant traits.
+                2. Pass trait to `NotebookApp`.
+            - If the argument is a trait of just `ServerApp` only
+                (i.e. the trait moved from `NotebookApp` to `ServerApp`):
```

Comment:
Just to clarify, take this construed example:
- If I start nbclassic, I want the server to serve it from port 8888
- I want all other jupyter servers to serve from port 9001

How would I configure this?

Reply:
I think you would want to launch two separate servers. Launch nbclassic like so:  

```
jupyter nbclassic --ServerApp.port=8888
```

then start a second server with
```
jupyter server --ServerApp.port=9001 --ServerApp.jpserver_extensions=[...]
```

Aliases can be used in both examples to shorten the CLI. You can also point each server to separate config files with these options set there too.

Reply:
Ah, so basically the answer is "this is not possible via the config system, please use command arguments". Sure.

Reply:
Oh, I see. I thought you were asking about running two servers *at the same time*. 

Right—if you have redundant config, the "last write wins". Currently, extensions are sorted alphabetically before they are loaded. So if two extensions configure the server, the last one in the list will override all others.

Reply:
> Oh, I see. I thought you were asking about running two servers at the same time.

I was, but without using command arguments :)